### PR TITLE
The PR for tsp, add test cases for Encode/Parameters/Versioning/Type

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issues#2738](https://github.com/Azure/autorest.java/issues/2738)

Added test cases for Encode/Parameters/Versioning/Type

### CODE REVIEW

File: typespec-tests/src/test/java/com/parameters/basic/BasicClientTests.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+public class BasicClientTests {
+    private final ExplicitBodyClient explicitBodyClient = new BasicClientBuilder().buildExplicitBodyClient();
+    private final ImplicitBodyClient implicitBodyClient = new BasicClientBuilder().buildImplicitBodyClient();
+
+    @Test
+    public void testBodyClient() {
+        explicitBodyClient.simple(new User("foo"));
+        implicitBodyClient.simple("foo");
+    }
+}
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/parameters/basic/BasicClientTests.java:18]: nit, newline in the end of the file

File: typespec-tests/src/test/java/com/versioning/added/AddedClientTests.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.azure.core.util.BinaryData;
+import com.versioning.added.models.EnumV1;
+import com.versioning.added.models.EnumV2;
+import com.versioning.added.models.ModelV1;
+import com.versioning.added.models.ModelV2;
+import org.junit.jupiter.api.Test;
+
+public class AddedClientTests {
+    private final AddedClient addedClient = new AddedClientBuilder()
+            .endpoint("http://localhost:3000").buildClient();
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/versioning/added/AddedClientTests.java:15]: Do we have to set the endpoint?

File: typespec-tests/src/test/java/com/type/model/flatten/FlattenClientTests.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.junit.jupiter.api.Test;
+
+public class FlattenClientTests {
+    private final FlattenClient flattenClient = new FlattenClientBuilder().buildClient();
+
+    @Test
+    public void testPutFlattenModel() {
+        flattenClient.putFlattenModel(new FlattenModel("foo", new ChildModel("bar", 10)));
+        flattenClient.putNestedFlattenModel(new NestedFlattenModel("foo", new ChildFlattenModel("bar", new ChildModel("test",10))));
+    }
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/model/flatten/FlattenClientTests.java:19]: Add a comment that the flatten is not implemented for data-plane.
(this test is for model class shape)

File: typespec-tests/src/test/java/com/versioning/added/AddedClientTests.java (hunk 2 above)
Comment (v-hongli1) [reply typespec-tests/src/test/java/com/versioning/added/AddedClientTests.java:15]: `reactor.core.Exceptions$ReactiveException: java.net.MalformedURLException: no protocol: /versioning/added/api-version:/interface-v2/v2` exception occurs without endpoint.

File: typespec-tests/src/test/java/com/type/model/flatten/FlattenClientTests.java (hunk 3 above)
Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/model/flatten/FlattenClientTests.java]: Fixed in the new version.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**